test/tests/functional/pbs_qrun.py

```diff
@@ -115,25 +115,26 @@ def test_qrun_hangs(self):
                 self.server.submit(j)
         self.logger.info("Submitted 500 jobs with different walltime")
 
+        now = int(time.time())
         self.server.manager(MGR_CMD_SET, SCHED,
                             {'scheduling': 'True'})
         self.server.manager(MGR_CMD_SET, SCHED,
                             {'scheduling': 'False'})
-        now = int(time.time())
-        try:
-            self.server.runjob(jobid=jid)
-        except PbsRunError as e:
-            # sched cycle started by qrun and failed as no free nodes
-            if err_msg in e.msg[0]:
-                self.logger.info(
-                    "No free nodes available. Checking for sched cycle")
-                try:
-                    self.scheduler.log_match("Starting Scheduling Cycle",
-                                             starttime=now,
-                                             max_attempts=10)
-                except PtlLogMatchError:
-                        self.assertFalse(True,
-                                         "Sched cycle not started by qrun")
-        except Exception as e:
-            if "Timed out" in str(e):
-                self.logger.info("Qrun hung: " + str(e))
+        pid = os.fork()
+        if pid == 0:
+            try:
+                self.server.runjob(jobid=jid)
+            except PbsRunError as e:
+                self.logger.info("Runjob throws error: " + e.msg[0])
+        else:
+            time.sleep(5)
+            try:
+                out = self.scheduler.log_match("Starting Scheduling Cycle",
+                                               allmatch=True,
+                                               starttime=now,
+                                               max_attempts=10)
+                if len(out) < 2:
+                    self.assertFalse(True,
```

Comment:
Use self.assertLess() instead of an if and assertFalse(True)

Reply:
Thanks @bhroam. I have modified the PTL. Added extra steps to handle the hung process.
@agrawalravi90 @bhroam Can you kindly review the latest commit "Incorporated Changes".